src/scripts/train/learn2code/love2code-3B.py

```diff
@@ -0,0 +1,301 @@
+"""
+Train a 3B for a first coding run. Run this script without any arguments to see usage info.
+"""
+
+from olmo_core.config import DType
+from olmo_core.distributed.parallel import DataParallelType
+from olmo_core.internal.experiment import CommonComponents, main, build_common_components, SubCmd, ExperimentConfig
+from olmo_core.internal.common import get_work_dir, get_root_dir
+from olmo_core.train.common import Duration
+from olmo_core.nn.transformer import TransformerConfig, TransformerDataParallelConfig
+from olmo_core.optim import AdamWConfig, OptimGroupOverride, OptimConfig
+from olmo_core.data import (
+    DataMixBase,
+    NumpyDataLoaderConfig,
+    NumpyDatasetConfig,
+    TokenizerConfig,
+    TokenizerName,
+)
+from olmo_core.train import TrainerConfig
+from olmo_core.train.callbacks import CheckpointerCallback, CometCallback, WandBCallback
+from typing import Callable, Dict, List, Tuple, Optional, cast
+import sys
+
+SEQUENCE_LENGTH = 2048 
+CHINCHILLA_5X_DURATION = Duration.tokens(36072678400 * 20 * 5)
```

Comment:
Is that the real number of params? Should be non-embedding params. There might be a way to get this programmatically.